packages/patterns/README.md

```diff
@@ -1,103 +1,32 @@
-
-The main export from the package is an `M` namespace object, for making a variety of Matchers (hence "M").
-
-`M` can also make _Guards_ that use Patterns to characterize dynamic behavior such as method argument/response signatures and promise awaiting. The [`@endo/exo`](https://www.npmjs.com/package/@endo/exo) package uses InterfaceGuards (each of which maps a collection of method names to their respective method guards) as the first level of defense for Exo objects against malformed input. For example:
 ```js
-const AsyncSerializerI = M.interface('AsyncSerializer', {
-  // This interface has a single method, which is async as indicated by M.callWhen().
-  // The method accepts a single argument, consumed with an implied `await` as indicated by M.await(),
-  // and the result of that implied `await` is allowed to fulfill to any value per M.any().
-  // The method result is a string as indicated by M.string(),
-  // which is inherently wrapped in a promise by the async nature of the method.
-  getStringOf: M.callWhen(M.await(M.any())).returns(M.string()),
-});
-const asyncSerializer = makeExo('AsyncSerializer', AsyncSerializerI, {
-  // M.callWhen() delays invocation of this method implementation
-  // while provided argument is in a pending state
-  // (i.e., it is a promise that has not yet settled).
-  getStringOf(val) { return String(val); },
-});
-
-const stringP = asyncSerializer.getStringOf(Promise.resolve(42n));
-isPromise(stringP); // => true
-await stringP; // => "42"
-```
-
-See [types.js](./src/types.js) for the definitions of these new types and (at typedefs `PatternMatchers` and `GuardMakers`) the methods of the exported `M` namespace object.
-
-## Invariants
-
-Any Passable value is a possibly-empty tree of `passStyleOf`-level containers (CopyArray, CopyRecord, CopyTagged) in which each node may be extended with an arbitrary number of non-container Passable leaves (an isolated non-container Passable is a sole leaf of an empty tree).
-If no leaf is a Capability (i.e., a Remotable or Promise), then the Passable value is Data --- it carries only immutable information, without any connection to external references or unforgeable identity.
+import { M, mustMatch } from '@endo/patterns';
 
-Guards do not yet exist as distinct kinds, so we ignore them for now. TODO: Expand this if kinds expand to include guards.
+const specimen = { foo: 3, bar: 4 };
 
-As mentioned above, `keyEQ` is pass-invariant: if passing `xa` from vatA to vatB arrives as `xb`, and likewise `ya` and `yb`, then `keyEQ(xa,ya)` iff `keyEQ(xb,yb)`. And because we do not wish to give Promises, Errors, or unrecognized CopyTagged values any useful pass-invariant equality, a Key may not include any of those.
+const expected = M.splitRecord(
+   { foo: M.number() },
+   { bar: M.string(), baz: M.number() },
+);
```

Comment:
Would it be correct to have this say:
```js
  { foo: M.number() }, // required properties
  { bar: M.string(), baz: M.number() }, // optional
```

so a newbie to `M` like me gets a hint as to why there are two arguments?